src/lib/mongoBackend/mongoQueryContext.cpp

```diff
@@ -310,12 +310,12 @@ HttpStatusCode mongoQueryContext
                        limit,
                        &limitReached,
                        countP);
-    reqSemGive(__FUNCTION__, "ngsi10 query request", reqSemTaken);
```

Comment:
This is not me, the merge with develop must have done this ...
I think.
Take a good look at it and decide whether we need the semaphore all the way to the end.
I am not so sure ...  It's not good to take the semaphore any more time than really needed ...